src/BesselFunctions/besselk.jl

```diff
@@ -614,3 +602,68 @@ end
         end
     )
 end
+
+# This is an expansion of the function
+#
+# f_0(v, x) = (x^v)*gamma(-v)  + (x^(-v))*gamma(v)
+#           = (x^v)*(gamma(-v) + (x^(-2*v))*gamma(v))
+#
+# around v ∼ 0. As you can see by plugging that second form into Wolfram Alpha
+# and getting an expansion back, this is actually a bivariate polynomial in
+# (v^2, log(x)). So that's how this is structured.
+@inline function f0_local_expansion_v0(v, x)
+    lx = log(x)
+    c0 = evalpoly(lx, (-1.1544313298030657,  -2.0))
+    c2 = evalpoly(lx, ( 1.4336878944573288,  -1.978111990655945, -0.5772156649015329,  -0.3333333333333333))
+    c4 = evalpoly(lx, (-0.6290784463642211,  -1.4584260788225176,   -0.23263776388631713, -0.32968533177599085, -0.048101305408461074, -0.016666666666666666))
+    evalpoly(v*v, (c0,c2,c4))/2
+end
+
+# This function assumes |v| < 1e-6 or 1e-7!
+# 
+# TODO (cg 2023/05/16 18:07): lots of micro-optimizations.
+function besselk_power_series_temme_basal(v::V, x::Float64) where{V}
+    max_iter = 50
+    T   = promote_type(V,Float64)
+    z   = x/2
+    zz  = z*z
+    fk  = f0_local_expansion_v0(v, x/2)
+    zv  = z^v
+    znv = inv(zv)
+    gam_1pv = GammaFunctions.gamma_near_1(1+v)
+    gam_1nv = GammaFunctions.gamma_near_1(1-v)
+    (pk, qk, _ck, factk, vv) = (znv*gam_1pv/2, zv*gam_1nv/2, one(T), one(T), v*v)
+    (out_v, out_vp1) = (zero(T), zero(T))
+    for k in 1:max_iter
+        # add to the series:
+        ck       = _ck/factk
+        term_v   = ck*fk
+        term_vp1 = ck*(pk - (k-1)*fk)
+        out_v   += term_v
+        out_vp1 += term_vp1
+        # check for convergence:
+        ((abs(term_v) < eps(T)) && (abs(term_vp1) < eps(T))) && break
+        # otherwise, increment new quantities:
+        fk     = (k*fk + pk + qk)/(k^2 - vv)
+        pk    /= (k-v)
+        qk    /= (k+v)
+        _ck   *= zz
+        factk *= k
+    end
+    (out_v, out_vp1/z)
+end
+
+function besselk_power_series_int(v, x::Float64)
+    v < zero(v) && return besselk_power_series_int(-v, x)
+    flv = Int(floor(v))
```

Comment:
I think it's usually easiest to just indiscrimantely have `v = abs(v)` at the top. I'm wondering how this line affects derivative information in anyway. As the zero order derivative is obviously zero from this line. Though we don't really need this check because we are checking at the top level but I think this is fine.

I think I generally like using the `modf` function and I think this would be a good fit for these type of problems.